values/time.go

```diff
@@ -67,7 +67,41 @@ func (t Time) Truncate(d Duration) Time {
 }
 
 func (t Time) Add(d Duration) Time {
-	return t + Time(d.Duration())
+	newT := t
+	if d.months > 0 {
+		// Retrieve the current date and increment the values
+		// based on the number of months.
+		ts := t.Time()
+		year, month, day := ts.Date()
+		year += int(d.months / 12)
```

Comment:
Shouldn't you subtract here instead for negative durations?

Reply:
Yea I'll add test cases for negative durations. I messed up and forgot to do that.